lib/puppet/provider/service/upstart.rb

```diff
@@ -26,6 +24,16 @@
            :status_exec  => "/sbin/status",
            :initctl => "/sbin/initctl"
 
+  # We only want to use upstart as our provider if the upstart daemon is running.
+  # This can be checked by running `initctl version --quiet` on a machine that has
+  # upstart installed.
+  confine :true => begin
```

Comment:
In case anyone's curious, the command methods are created before this confine block per https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider.rb#L167 and https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider.rb#L217-L224. That's why we place it after.